### PR TITLE
feat(HACBS-1725): improve gosec pr check

### PR DESCRIPTION
**Pull request by mmalina**
This will add a step to upload sarif file generated by gosec to Github using the upload-sarif action.

That way if gosec finds issues in the code,
Github UI will show them inline in the Files
tab of a PR.

### CODE REVIEW

**Review by happybhati — DISMISSED** [review]

File: .github/workflows/gosec.yaml
Comment (mmalina) [inline .github/workflows/gosec.yaml:26]: They have the `-no-fail` in [the documentation](https://github.com/securego/gosec#integrating-with-code-scanning), but I'm not completely sure about this. I think it would make sense to still fail here even with the warnings/errors shown in the UI. I wonders what others think.

File: main.go
Comment (mmalina) [reply main.go:106]: This is just to show what it looks like. I will remove this change before merging.

**Review by davidmogar — APPROVED** [review]

File: .github/workflows/gosec.yaml
Comment (johnbieren) [reply .github/workflows/gosec.yaml:26]: I think we should still fail

**Comment by johnbieren** [comment]
I'm good to approve once the example fail is removed

File: .github/workflows/gosec.yaml
Comment (mmalina) [reply .github/workflows/gosec.yaml]: OK, thanks for the input. I tried removing the `-no-fail`. But the problem now is that if the `Run Gosec Security Scanner` step fails, the rest of the job is skipped (including the `Upload SARIF file` step). So the warning in the UI will not show up.

Comment (mmalina) [reply .github/workflows/gosec.yaml]: https://github.com/redhat-appstudio/release-service/actions/runs/5728029661/job/15521693572?pr=228

```
Run if [ "failure" != "success" ]
Gosec failed. The errors and warnings should be displayed in the PR.
Error: Process completed with exit code 1.
```

File: .github/workflows/gosec.yaml
Comment (johnbieren) [inline .github/workflows/gosec.yaml:42]: extra new line

**Review by johnbieren — COMMENTED** [review]
lgtm once `main.go` modifications are removed

File: .github/workflows/gosec.yaml
Comment (mmalina) [reply .github/workflows/gosec.yaml]: fixed, thanks

**Review by johnbieren — APPROVED** [review]

File: main.go
Comment (mmalina) [reply main.go]: This change is removed now.

**Comment by mmalina** [comment]
The testing commit is removed now, so this is ready to be merged. As always, we need the e2e run to pass. So if anybody sees it passed before I do, feel free to merge this.

**Pull request merged**
